crates/revm/revm-inspectors/src/stack/maybe_owned.rs

```diff
@@ -0,0 +1,209 @@
+use revm::{
+    interpreter::{CallInputs, CreateInputs, Gas, InstructionResult, Interpreter},
+    primitives::{db::Database, Bytes, B160, B256},
+    EVMData, Inspector,
+};
+use std::{
+    cell::{Ref, RefCell},
+    rc::Rc,
+};
+
+/// An [Inspector] that is either owned by an individual [Inspector] or is shared as part of a
+/// series of inspectors in a [InspectorStack](crate::stack::InspectorStack).
+///
+/// Caution: if the [Inspector] is _stacked_ then it _must_ be called first.
+#[derive(Debug)]
+pub enum MaybeOwnedInspector<INSP> {
```

Comment:
Where do you envision this kind of inspector to be used?

Reply:
either only used for one inspector like tracing or as part of stack